Lombiq.NodeJs.Extensions/scripts/validate-config.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @summary Validates the Node.js Extensions configuration object.
+ * @description This script validates a configuration object against the format provided in @example.
+ * @example "nodejsExtensions": {
+ *            "assetsToCopy": [
+ *              {
+ *                "sources": [ "Assets/Images" ],
+ *                "target": "wwwroot/images"
+ *              },
+ *              {
+ *                "sources": [
+ *                  "node_modules/marvelous/dist"
+ *                  "node_modules/wonderful/bin"
+ *                ],
+ *                "pattern": "*",
+ *                "target": "wwwroot"
+ *              }
+ *              // more groups, if needed
+ *            ],
+ *            "scripts": {
+ *              "source": "CustomJsFolder",
+ *              "target": "wwwCustomRoot/jay-es"
+ *            },
+ *            "styles": {
+ *              "source": "NonDefaultScssFolder",
+ *              "target": "wwwwNonDefaultCssFolder"
+ *            }
+ *          }
+ */
+function logError(message) { process.stderr.write(`ERROR: ${message}\n`); }
+
+function validateAssetGroupsAndLogErrors(assetConfig) {
+    if (!Array.isArray(assetConfig)) {
+        logError('The configuration must be an array of asset groups of the form: ' +
+            '{ sources: Array<string>, pattern: string [optional], target: string }.');
+        return false;
+    }
+
+    const errors = [];
+
+    assetConfig.forEach((assetGroup) => {
+        if (!Array.isArray(assetGroup.sources)) {
+            errors.push('sources must be an array of strings');
+        }
+        if (assetGroup.pattern && typeof assetGroup.pattern !== 'string') {
+            errors.push('pattern must be a glob string');
+        }
+        if (typeof assetGroup.target !== 'string') {
+            errors.push('target must be a string');
+        }
+        if (errors.length > 0) {
+            logError(`Invalid asset group: ${JSON.stringify(assetGroup)}: ${errors.join(', ')}.`);
+        }
+    });
+
+    return errors.length === 0;
+}
+
+function validateSimpleGroupAndLogErrors(group) {
+    const errors = [];
+    if (typeof group.source !== 'string') {
+        errors.push('source must be a string');
+    }
+    if (typeof group.target !== 'string') {
+        errors.push('target must be a string');
+    }
+    if (errors.length > 0) {
+        logError(`Invalid group: ${JSON.stringify(group)}: ${errors.join(', ')}.`);
+    }
+    return errors.length === 0;
+}
+
+function validateAndLogErrors(nodejsExtensionsConfig) {
+    return Object.entries(nodejsExtensionsConfig).every(([key, group]) => {
```

Comment:
I didn't realize you can destructure an array in the function parameter. This is amazing!